src/lib.rs

```diff
@@ -135,11 +164,45 @@ impl<'object, Endian> DebugInfo<'object, Endian>
         let debug_str = file.get_section(".debug_str").unwrap_or(&[]);
         let debug_str = gimli::DebugStr::<Endian>::new(debug_str);
 
+        // Collect all the function names so we can deal with inlined calls
+        let mut func_names = if with_functions {
+            Some(HashMap::new())
+        } else {
+            None
+        };
+        if let Some(ref mut func_names) = func_names {
+            let mut headers = debug_info.units();
+            while let Some(header) = headers.next()
+                .map_err(|_| MappingError::MissingDebugInfo("couldn't get DIE header"))? {
+                let abbrev = header.abbreviations(debug_abbrev).expect("Fail");
```

Comment:
This is a left-over from the original `addr2line` code as it was in gimli ;)
I'm planning on doing a subsequent PR that cleans up error handling throughout the code, but will modify this one while I'm here.